### PR TITLE
Feat(identityProviders): useIsSignedIn Hook

### PR DESCRIPTION
**Pull request by AhmetAhunbayAWS**
#### Description of changes

Creates a new react hook that gives customers access to whether or not the end user is signed in. Creates a tester file and updates index files to expose the hook to the customer. Adds new hook to jest snapshot file. This react hook is decoupled from the existing Authenticator.

#### Description of how you validated changes

I created a test file to go over possible cases and ran yarn test.

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-core/src/hooks/useIsSignedIn.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      });
+  }, []);
+
+  useEffect(() => {
+    const listener: HubCallback = ({ payload }) => {
+      if (payload.event === 'signedIn') {
+        setIsSigned(true);
+      } else if (payload.event === 'signedOut') {
+        setIsSigned(false);
+      }
```
Comment (thaddmt) [inline packages/react-core/src/hooks/useIsSignedIn.ts:34]: Does this work with other auth hub events? I recall there being events like `tokenRefresh_failure `, does `signedOut` get called after those events and does the hook successfully show signed in status for something like that?

**Review by thaddmt — COMMENTED** [review]
LGTM other than a question!

File: packages/react-core/src/hooks/useIsSignedIn.ts (hunk 1 above)
Comment (AhmetAhunbayAWS) [reply packages/react-core/src/hooks/useIsSignedIn.ts:34]: I think yes, I have to catch that event as well. 

If you look at [useAuth](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/hooks/useAuth.ts), it seems like tokenRefresh_failure and signIn_failure have the same functionality, which is different from signedOut and doesn't update the useAuth user state. 

But if you look at [defaultHubHandler](https://github.com/aws-amplify/amplify-ui/blob/main/packages/ui/src/helpers/authenticator/defaultAuthHubHandler.ts), both tokenRefresh_failure and signedOut send out the same event that triggers a signOutActor in the state machine. But I did not find any references to this defaultHubHandler anywhere else in the codebase. 

Also, I asked someone on JS and they said they do not send a signedOut event along with the tokenRefresh_failure event.

**Review by thaddmt — APPROVED** [review]

File: packages/react-core/src/hooks/useIsSignedIn.ts (hunk 1 above)
Comment (thaddmt) [reply packages/react-core/src/hooks/useIsSignedIn.ts:34]: Sounds good to me

**Review by axelEandrews — APPROVED** [review]
Looks good to me!

**Pull request closed without merging**